flang/lib/Optimizer/HLFIR/Transforms/OptimizedBufferization.cpp

```diff
@@ -462,16 +474,20 @@ ElementalAssignBufferization::findMatch(hlfir::ElementalOp elemental) {
   // the incoming expression
   match.array = match.assign.getLhs();
   mlir::Type arrayType = mlir::dyn_cast<fir::SequenceType>(
-      fir::unwrapPassByRefType(match.array.getType()));
-  if (!arrayType)
+      hlfir::getFortranElementOrSequenceType(match.array.getType()));
+  if (!arrayType) {
```

Comment:
`getFortranElementOrSequenceType` will return the element type if the passed type is not an array. I think this should be

```suggestion
  if (!mlir::isa<fir::SequenceType>(arrayType)) {
```

Reply:
Thanks for the review, Tom! There is a `dyn_cast<fir::SequenceType>` above, which would set `arrayType` to null if it was not an array, so I think it should be okay.  Anyway, I will move this change into another PR.